lib/protobuf/enum.rb

```diff
@@ -120,7 +127,8 @@ def self.enum_for_name(name)
     #   Enums, the first enum defined will be returned.
     #
     def self.enum_for_tag(tag)
-      enums_for_tag(tag).first
+      value = mapped_enums[tag.to_i]
+      value ? value.first : nil
```

Comment:
is it an error to have more than one enum value? Should this pre-emptively raise on that case instead? (likely out of scope for this PR)